### PR TITLE
Staking Proxy Unit Tests + Fallback Reverts if No Staking Contract

### PR DESCRIPTION
**Pull request by hysz**
## Description

Staking Proxy unit tests + Staking Proxy fallback reverts if no staking contract is attached.

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/staking/test/unit_tests/staking_proxy_test.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+            // Cache existing staking contract and attach a new one
+            const initStakingContractAddress = await testProxyContract.stakingContract.callAsync();
+            const txReceipt = await testProxyContract.attachStakingContract.awaitTransactionSuccessAsync(
+                testContract2.address,
+                { from: authorizedAddress },
+            );
+
+            // Validate `ContractAttachedToProxy` event
+            expect(txReceipt.logs.length).to.be.gte(1);
+            const contractAttachedEvent: StakingProxyStakingContractAttachedToProxyEventArgs = (txReceipt
```
Comment (abandeali1) [inline contracts/staking/test/unit_tests/staking_proxy_test.ts]: This should be `const contractAttachedEvent = (txReceipt.logs[0] as LogWithDecodedArgs<StakingProxyStakingContractAttachedToProxyEventArgs>).args;` in order to be consistent with all of the other tests.

Comment (mzhu25) [reply contracts/staking/test/unit_tests/staking_proxy_test.ts]: we should adopt the convention of using `verifyEvents` from `test-utils` whenever we want to check event fields

File: contracts/staking/test/unit_tests/staking_proxy_test.ts [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+            ];
+            const tx = testProxyContract.batchExecute.callAsync(calls);
+            const expectedError = 'Goodbye, World!';
+            return expect(tx).to.revertWith(expectedError);
+        });
+    });
+
+    describe('assertValidStorageParams', () => {
+        const validStorageParams = {
+            epochDurationInSeconds: new BigNumber(5 * 24 * 60 * 60), // 5 days
```
Comment (mzhu25) [inline contracts/staking/test/unit_tests/staking_proxy_test.ts]: nit: we should throw `24 * 60 * 60` into `constants`

File: contracts/staking/test/unit_tests/staking_proxy_test.ts (hunk 1 above)
Comment (hysz) [reply contracts/staking/test/unit_tests/staking_proxy_test.ts]: @mzhu25 this is great!

**Pull request merged**